sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -0,0 +1,437 @@
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<PathInfo>> flushWithResponse(long position, boolean retainUncommittedData, boolean close,
+        PathHttpHeaders httpHeaders, DataLakeRequestConditions accessConditions, Context context) {
+
+        httpHeaders = httpHeaders == null ? new PathHttpHeaders() : httpHeaders;
+        accessConditions = accessConditions == null ? new DataLakeRequestConditions() : accessConditions;
+
+        LeaseAccessConditions lac = new LeaseAccessConditions().setLeaseId(accessConditions.getLeaseId());
+        ModifiedAccessConditions mac = new ModifiedAccessConditions()
+            .setIfMatch(accessConditions.getIfMatch())
+            .setIfNoneMatch(accessConditions.getIfNoneMatch())
+            .setIfModifiedSince(accessConditions.getIfModifiedSince())
+            .setIfUnmodifiedSince(accessConditions.getIfUnmodifiedSince());
+
+        return this.dataLakeStorage.paths().flushDataWithRestResponseAsync(null, position, retainUncommittedData, close,
+            (long) 0, null, httpHeaders, lac, mac, context)
+            .map(response -> new SimpleResponse<>(response, new PathInfo(response.getDeserializedHeaders().getETag(),
+                response.getDeserializedHeaders().getLastModified())));
+    }
+
+    /**
+     * Reads the entire file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.read}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob">Azure Docs</a></p>
+     *
+     * @return A reactive response containing the file data.
+     */
+    public Flux<ByteBuffer> read() {
+        try {
+            return readWithResponse(null, null, null, false)
+                .flatMapMany(FileDownloadAsyncResponse::getValue);
+        } catch (RuntimeException ex) {
+            return fluxError(logger, ex);
+        }
+    }
+
+    /**
+     * Reads a range of bytes from a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.readWithResponse#FileRange-DownloadRetryOptions-DataLakeRequestConditions-boolean}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob">Azure Docs</a></p>
+     *
+     * @param range {@link FileRange}
+     * @param options {@link DownloadRetryOptions}
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @param rangeGetContentMD5 Whether the contentMD5 for the specified file range should be returned.
+     * @return A reactive response containing the file data.
+     */
+    public Mono<FileDownloadAsyncResponse> readWithResponse(FileRange range, DownloadRetryOptions options,
+        DataLakeRequestConditions accessConditions, boolean rangeGetContentMD5) {
```

Comment:
Request conditions should go last

Reply:
This copies the order of parameters for blobs, should I still change it?

Reply:
No :(

---
In reply to: [340815839](https://github.com/Azure/azure-sdk-for-java/pull/6071#discussion_r340815839) [](ancestors = 340815839)